pkg/pillar/cmd/volumemgr/handleconfig.go

```diff
@@ -108,6 +108,12 @@ func vcCreate(ctx *volumemgrContext, objType string, key string,
 		initStatus.LastUse = time.Now()
 		initStatus.PreReboot = false
 		publishVolumeStatus(ctx, initStatus)
+		if !initStatus.DownloadOrigin.HasVerifierRef {
+			log.Infof("vcCreate: Adding verified reference for initVolStatus: %s", initStatus.DisplayName)
+			AddOrRefcountVerifyConfig(ctx, initStatus)
```

Comment:
You can't have have this function create a VerifyImageConfig because handleCreate in the verifier would look for a file in the pending directory, which doesn't exist.

Reply:
Instead you need to check that there is either a verifyImageStatus or PersistImageStatus before adding a reference to the VerifyImageConfig  (I assume that if there is a VerifyImageStatus there is also a VerifyImageConfig).

Reply:
But in this case, we will have a verified image and a persistImageStatus already. So in handleCreate, verifier checks if it has a persistImageStatus for that sha, and if it's present, then verifier returns a VERIFIED status.

Reply:
Do we? We have an image in /persist/img hence an initial VolumeStatus but no guarantee that we have anything in /persist/downloads/*/verified/

If it is a nice sunny day we might, but the code can not assume that. Its quite likely that it will break when updating EVE even if it works in the new version of EVE.

Reply:
Added a check.